diff --git a/ImageMagick-7.1.0-62/MagickCore/exception.c b/ImageMagick-7.1.0-62/MagickCore/exception.c
index 57f60aac..e6e414b2 100644
--- a/ImageMagick-7.1.0-62/MagickCore/exception.c
+++ b/ImageMagick-7.1.0-62/MagickCore/exception.c
@@ -525,16 +525,16 @@ MagickExport char *GetExceptionMessage(const int error)
     exception[MagickPathExtent];
 
   *exception='\0';
-#if defined(MAGICKCORE_HAVE_STRERROR_R)
-#if !defined(MAGICKCORE_STRERROR_R_CHAR_P)
+  /* #if defined(MAGICKCORE_HAVE_STRERROR_R)
+     #if !defined(MAGICKCORE_STRERROR_R_CHAR_P) */
   (void) strerror_r(error,exception,sizeof(exception));
-#else
+  /* #else
   (void) CopyMagickString(exception,strerror_r(error,exception,
     sizeof(exception)),sizeof(exception));
 #endif
 #else
   (void) CopyMagickString(exception,strerror(error),sizeof(exception));
-#endif
+  #endif */
   return(ConstantString(exception));
 }
 
diff --git a/make/libmagickcore-7.mk b/make/libmagickcore-7.mk
index 87092e85..5f2c4d1b 100644
--- a/make/libmagickcore-7.mk
+++ b/make/libmagickcore-7.mk
@@ -369,6 +369,7 @@ ifeq ($(LIBLCMS2_ENABLED),true)
     LOCAL_STATIC_LIBRARIES += liblcms2
 endif
 
+LOCAL_CFLAGS += -std=gnu89
 
 ifeq ($(STATIC_BUILD),true)
     include $(BUILD_STATIC_LIBRARY)
diff --git a/make/libxml2.mk b/make/libxml2.mk
index ace5dbc4..5d3b55a8 100644
--- a/make/libxml2.mk
+++ b/make/libxml2.mk
@@ -20,6 +20,7 @@ LOCAL_CFLAGS := \
     -Wno-tautological-pointer-compare \
     -Wno-unused-function \
     -Wno-unused-parameter \
+    -Wno-error=unused-but-set-variable \
 
 ifeq ($(LIBLZMA_ENABLED),true)
     LOCAL_CFLAGS += -DLIBXML_LZMA_ENABLED=NO
